### ISSUE TITLE
Importer returning contents and file no longer retain absolute paths

### ISSUE THREAD

**Issue by gmac** [sass/node-sass#1219]
There is a significant breaking API change in the way imported file contents reference subsequent imports between versions 3.3.3 and 3.4.0.

In **v3.3.3**, absolute file references provided with imported contents would be passed along for resolving subsequent imports. Example behavior in v3.3.3:

```javascript
// @import "a"
done({
  file: '/Users/me/sass/lib/a.scss',
  contents: '@import "b"'
});
```

_Next importer call for `@import "b"`_:

```javascript
function(uri, previousUri, done) {
   // uri: "b",
   // previousUri: "/Users/me/sass/lib/a.scss"
}
```

In **v3.4.0**, absolute file references are apparently discarded, and only the previous slug is passed along to subsequent importer calls. Example behavior in v3.4.0:

```javascript
// @import "a"
done({
  file: '/Users/me/sass/lib/a.scss',
  contents: '@import "b"'
});
```

_Next importer call for `@import "b"`_:

```javascript
function(uri, previousUri, done) {
   // uri: "b",
   // previousUri: "a"
}
```

This is a significant detriment to content importers that rely on the previously resolved content URI for file mapping, and a major enough change that I would have expected it to be part of a major version bump if intentional. If this change _is_ intentional, I'd be curious about the rationale for removing this exceptionally valuable piece of data from the implementation. Without this data, importers now need to maintain a complex external dependency mapping table that can –at best– still only guess upon a previous slug's mapped file.

**Comment by mgreter** [comment]
I guess https://github.com/sass/node-sass/blob/master/src/binding.cpp#L12 is the culprit, should probably be `sass_import_get_imp_path`. The names have changed in libsass API and it looks like this got mixed up on the node sass side. Looks like node-sass could use some more unit tests.

**Comment by gmac** [comment]
`Looks like node-sass could use some more unit tests.` :+1:

**Comment by mgreter** [comment]
@gmac I guess their willing to accept Pull Requests in that direction :wink:

**Comment by gmac** [comment]
Noted, and considered.

**Comment by xzyfer** [comment]
This was fixed in https://github.com/sass/node-sass/pull/1220. Will land in v3.4.2.

**Comment by Rush** [comment]
I am experiencing this problem as well, waiting for release then!

**Comment by crypticsymbols** [comment]
+1 eagerly awaiting this

**Comment by xzyfer** [comment]
I believe this is fixed in v3.4.2. Please open if the issue persists.
